app/src/main/java/com/ably/tracking/demo/publisher/ably/log/LocationLogger.kt

```diff
@@ -0,0 +1,76 @@
+package com.ably.tracking.demo.publisher.ably.log
+
+import com.ably.tracking.Location
+import com.ably.tracking.LocationUpdate
+import com.ably.tracking.publisher.LocationHistoryData
+import com.google.gson.Gson
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+class LocationLogger(
+    private val fileWriter: LogFileWriter,
+    private val fileManager: FileManager,
+    private val gson: Gson
+) {
+
+    companion object {
+        private const val LOG_TIME_FORMATTER_PATTERN = "HH:mm:ss"
+        private const val FILE_NAME_FORMATTER_PATTERN = "dd.MM_HH:mm:ss"
```

Comment:
That's an unusual pattern :thinking: Why there are both `.` and `_` in it? Is this a requirement of some sort? I would either choose `.` or `_` and not mix them together if we don't have strong reasons :wink: